### PR TITLE
[depends: #1333] pointers now come with a width

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Comment by tautschnig** [comment]
Marking do-not-merge until 1333 is CI'ed and merged.

File: src/ansi-c/type2name.cpp [hunk 1]
```diff
@@ -134,9 +134,12 @@ static std::string type2name(
   else if(type.id()==ID_natural)
     result+='N';
   else if(type.id()==ID_pointer)
-    result+='*';
-  else if(type.id()==ID_reference)
-    result+='&';
+  {
+    if(type.get_bool(ID_C_reference))
```
Comment (thk123) [inline src/ansi-c/type2name.cpp]: This feels like it shouldn't be in the ansi-c front end?

**Review by thk123 — COMMENTED** [review]
Seems fine in principle - though I am not very familiar with a lot of code this touches. I have created a test-gen submodule bump to check it works with test-gen, once that has passed I will approve.

**Review by thk123 — APPROVED** [review]
Works with the TG repo: diffblue/test-gen#1083 though needs to work with this repo!

**Review by thk123 — COMMENTED** [review]
Removing approval as just spotted fails CI on this repo!

**Comment by tautschnig** [comment]
Not sure what exactly happened here, but it seems typedefs are no longer being used in the output.

**Comment by kroening** [comment]
I've made the test more generous; the question whether or not the typedef is used here is not what's being tested.

**Comment by tautschnig** [comment]
> I've made the test more generous; the question whether or not the typedef is used here is not what's being tested.

While I agree that this isn't what the regression test needs to focus on, the fact that it broke/changed is alarming. This suggests that typedef information is being lost.

**Comment by thk123** [comment]
> The fact that it broke/changed is alarming.

I agree - it suggests to me (since expr2c hasn't changed) that at some point some of the type information is being dropped. While this isn't important for typedef's, it is fairly hard to know if it is loosing other, important information. That said - I've had another look through the code and can't spot anywhere that looks wrong.

**Pull request merged**